builtin/plugins/dposv3/dpos.go

```diff
@@ -197,6 +198,40 @@ func (c *DPOS) Init(ctx contract.Context, req *InitRequest) error {
 		params.DowntimePeriod = defaultDowntimePeriod
 	}
 
+	chainID := "default"
+	if req.ChainId != "" {
+		chainID = req.ChainId
+	}
```

Comment:
Should be able to use `ctx.Block().ChainID` here, no?

Reply:
Done, ChainID was empty in init block but not it is fixed.